File: src/main/java/de/com/fdm/main/Application.java
package de.com.fdm.main;

import de.com.fdm.redis.RedisListener;
import de.com.fdm.tmi.Reader;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.event.ApplicationReadyEvent;
import org.springframework.context.event.EventListener;


@SpringBootApplication(scanBasePackages = "de.com.fdm.*")
public class Application {
    private final Reader reader;
    private final RedisListener redisListener;

    @Autowired
    public Application(Reader reader, RedisListener redisListener) {
        this.reader = reader;
        this.redisListener = redisListener;
    }

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }

    @EventListener(ApplicationReadyEvent.class)
    private void joinSavedChannels() {
        reader.joinSavedChannels();

        redisListener.setUpCallback();
    }
}

File: src/main/java/de/com/fdm/main/Application.kt
package de.com.fdm.main

import de.com.fdm.core.LoadBalancer
import de.com.fdm.core.RedisListener
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.event.ApplicationReadyEvent
import org.springframework.boot.runApplication
import org.springframework.context.event.EventListener
import org.springframework.scheduling.annotation.EnableScheduling


@EnableScheduling
@SpringBootApplication(scanBasePackages = ["de.com.fdm.*"])
class Application @Autowired constructor(
    private val redisListener: RedisListener,
    private val loadBalancer: LoadBalancer,
) {
    private val log = LoggerFactory.getLogger(Application::class.java)

    @EventListener(ApplicationReadyEvent::class)
    fun init() {
        loadBalancer.initReaders()
        redisListener.start()
    }
}

fun main(args: Array<String>) {
    runApplication<Application>(*args)
}
